scripts/take_down_deployment_be_careful.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+
+import boto3
+
+lambda_client = boto3.client('lambda')
+api_gateway_client = boto3.client('apigateway')
+batch_client = boto3.client('batch')
+
+cdn_non_prod_template = "matrix.$DEPLOYMENT_STAGE.data.humancellatlas.org"
+cdn_prod = "matrix.data.humancellatlas.org"
+lambda_name_templates = [
+    "matrix-service-api-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-driver-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-mapper-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-worker-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-reducer-$DEPLOYMENT_STAGE"
+]
+job_queue_template = "dcp-matrix-converter-queue-$DEPLOYMENT_STAGE"
+
+
+def main(args):
+    # Take down API Gateway
+    if args.environment == "prod":
+        cdn = cdn_prod
+    else:
+        cdn = cdn_non_prod_template.replace("$DEPLOYMENT_STAGE", args.environment)
+    api_gateway_client.delete_domain_name(
+        domainName=cdn
```

Comment:
Is it possible for someone to continue to access the API without the custom domain name?

Reply:
I don't think this is possible. Custom domain name apigateway record is the link between an api gateway stage and route 53. As far as I know, you can't access an api gateway stage without this routing setup

Reply:
^I may be wrong here. I didn't know about api gateway invoke urls... Looking into this

Reply:
As @mckinsel brought up, the API is still available via the generated invoke URL `<stage_id>.execute-api.us-east-1.amazonaws.com`  We could attach a [Resource Policy](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-control-access-policy-language-overview.html) to the API denying `invoke` if we want to shut this down as well.